File: app/src/main/java/ai/elimu/content_provider/ui/storybook/StoryBooksViewModel.java
package ai.elimu.content_provider.ui.storybook;

import android.util.Log;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

public class StoryBooksViewModel extends ViewModel {

    private MutableLiveData<String> text;

    public StoryBooksViewModel() {
        text = new MutableLiveData<>();
        text.setValue("StoryBooksViewModel");
    }

    public LiveData<String> getText() {
        Log.i(getClass().getName(), "getText");

        return text;
    }
}

File: app/src/main/java/ai/elimu/content_provider/ui/storybook/StoryBooksViewModel.kt
package ai.elimu.content_provider.ui.storybook

import android.util.Log
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

class StoryBooksViewModel : ViewModel() {
    private val text = MutableLiveData<String>()

    init {
        text.value = "StoryBooksViewModel"
    }

    fun getText(): LiveData<String> {
        Log.i(javaClass.name, "getText")

        return text
    }
}
